### PR TITLE
Run rundoc on new PRs to check that the documents in src is can still run correctly

### PR DESCRIPTION
**Pull request by lilacstella**
Add another github action config to use the rundoc set up as a CI test.

### CODE REVIEW

**Comment by lilacstella** [comment]
This configuration triggers the test run on every new PR to main in this repo, there is a way to stop the run by checking if there are changes but it is [somewhat hacky](https://github.com/orgs/community/discussions/25669).

File: .github/workflows/rundoc-ci.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: rundoc-ci-test
+on:
+  pull_request:
+    branches:
+      - main
+jobs:
+  test-rundoc:
+    runs-on: pub-hk-ubuntu-24.04-ip
```
Comment (colincasey) [inline .github/workflows/rundoc-ci.yml:8]: This repo is public, why not use one of the public GHA runners like `ubuntu-latest`?

File: .github/workflows/rundoc-ci.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    steps:
+      - name: Get token for GH application (Linguist)
```
Comment (colincasey) [inline .github/workflows/rundoc-ci.yml:10]: We should be able to do the git checkout on this repo without a token, right? If so, drop this step.

File: .github/workflows/rundoc-ci.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        uses: actions/create-github-app-token@v1
+        id: generate-token
+        with:
+          app-id: ${{ vars.LINGUIST_GH_APP_ID }}
+          private-key: ${{ secrets.LINGUIST_GH_PRIVATE_KEY }}
+
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+          token: ${{ steps.generate-token.outputs.token }}
```
Comment (colincasey) [inline .github/workflows/rundoc-ci.yml:20]: I don't think the token is needed here.

**Review by colincasey — APPROVED** [review]

**Pull request merged**